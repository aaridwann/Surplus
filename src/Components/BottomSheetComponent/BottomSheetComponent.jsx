import React, { useEffect, useRef } from 'react';
import BottomSheet from 'react-native-gesture-bottom-sheet';
import { scaleHeight } from '../../Utils/Scale/Scale';
import PropTypes from 'prop-types';

const BottomSheetComponent = ({ children, show }) => {
	const bottomSheetRef = useRef();

	useEffect(() => {
		if (show) {
			bottomSheetRef.current.show();
		}
	}, [show]);

	return (
		<BottomSheet
			hasDraggableIcon
			ref={bottomSheetRef}
			height={scaleHeight(400)}
			testID="bottomSheet"
		>
			{children}
		</BottomSheet>
	);
};

BottomSheetComponent.propTypes = {
	children: PropTypes.node,
	show: PropTypes.bool,
};

export default BottomSheetComponent;
