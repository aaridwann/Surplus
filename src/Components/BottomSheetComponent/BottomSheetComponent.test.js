import React from 'react';
import { render } from '@testing-library/react-native';
import BottomSheetComponent from './BottomSheetComponent';
import { Text } from 'react-native';

describe('BottomSheetComponent', () => {
	test('Snapshot', () => {
		const { getByText, toJSON } = render(
			<BottomSheetComponent show={true}>
				<Text>Content</Text>
			</BottomSheetComponent>
		);

		const contentElement = getByText('Content');

		expect(contentElement).toBeDefined();
		expect(toJSON()).toMatchSnapshot();
	});
});
