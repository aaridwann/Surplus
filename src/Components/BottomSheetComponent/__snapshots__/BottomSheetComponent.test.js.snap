// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheetComponent Snapshot 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        },
        Object {
          "backgroundColor": "#25252599",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      collapsable={false}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#F3F3F3",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 0,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "height": 6,
                "margin": 10,
                "marginBottom": 0,
                "width": 40,
              },
              undefined,
              Object {
                "backgroundColor": "#A3A3A3",
              },
            ]
          }
        />
      </View>
      <Text>
        Content
      </Text>
    </View>
  </View>
</Modal>
`;
